### PR TITLE
TS common-adapters subfolders, misc

### PR DESCRIPTION
**Pull request by jakob223**
- [x] finish fixing `Text`

 - [x] `markdown/`
 - [x] `__test__/`
 - [x] `background-repeat-box/`
 - [x] `confirm-modal/`
 - [x] `floating-box/`
 - [x] `floating-menu/`
 - [x] `header-hoc/`
 - [x] `name-with-icon/`
 - [x] `overlay/`
 - [x] `usernames/`

Issue: DESKTOP-9814

### CODE REVIEW

**Comment by jakob223** [comment]
oops I meant to make this a draft PR..

File: shared/common-adapters/icon.native.tsx [hunk 1]
```diff
@@ -13,28 +12,28 @@ const Kb = {
...
-  opacity?: boolean,
-  sizeType: SizeType,
-  style?: Styles.StylesCrossPlatformWithSomeDisallowed<DisallowedStyles>,
-  type: IconType,
-|}
+type TextProps = {
+  children: React.ReactNode
+  color?: Styles.Color
+  fontSize?: number
+  onClick?: (event: React.SyntheticEvent) => void | null
```
Comment (buoyad) [inline shared/common-adapters/icon.native.tsx]:
```suggestion
  onClick?: ((event: React.SyntheticEvent) => void) | null
```

File: shared/common-adapters/icon.native.tsx [hunk 2]
```diff
@@ -13,28 +12,28 @@ const Kb = {
...
 
   // we really should disallow reaching into style like this but this is what the old code does.
   // TODO change this
-  // $FlowIssue
   const pStyle: any = p.style
 
   const color =
     p.color ||
-    // $FlowIssue flow is correct but we do actually pass in this color sometimes. TODO remove this
+    // @ts-ignore flow is correct but we do actually pass in this color
```
Comment (buoyad) [inline shared/common-adapters/icon.native.tsx]: `flow` -> `typescript`

File: shared/common-adapters/markdown/react.tsx [hunk 3]
```diff
@@ -286,7 +289,20 @@ const reactComponentsForMarkdownType = {
       json={node.content}
       key={state.key}
       allowFontScaling={state.allowFontScaling}
-      message={state.markdownMeta?.message ?? undefined}
+      message={
```
Comment (buoyad) [inline shared/common-adapters/markdown/react.tsx]: This can be a lot shorter

File: shared/common-adapters/text.native.tsx [hunk 4]
```diff
@@ -151,7 +150,12 @@ function _getStyle(
     ...textDecoration,
   }
 }
-function getStyle(type: TextType, negative?: boolean, lineClampNum?: ?number, clickable?: ?boolean) {
+function getStyle(
```
Comment (buoyad) [inline shared/common-adapters/text.native.tsx]: this is exported so this and desktop's version should have the same API

File: shared/common-adapters/text.d.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  isTerminal?: true
+}
+
+declare class Text extends React.Component<Props> {
+  highlightText: () => void
+}
+
+declare function getStyle(
+  type: TextType,
+  backgroundMode?: Background | null,
```
Comment (buoyad) [inline shared/common-adapters/text.d.ts]: add `selectable` here

File: shared/common-adapters/text.desktop.tsx [hunk 6]
```diff
@@ -115,10 +114,10 @@ function fastGetStyle(
 // Only used by external components
 function externalGetStyle(
```
Comment (buoyad) [inline shared/common-adapters/text.desktop.tsx]: did you check usages of this?

Comment (jakob223) [reply shared/common-adapters/text.desktop.tsx]: I looked for and couldn't find any, IIRC. I was confused at this situation, I would totally believe they're there and I missed them.

Comment (buoyad) [reply shared/common-adapters/text.desktop.tsx]: it's exported as `getStyle`, `search/user-input/index` has one for example. looks like most are in common adapters

**Comment by MarcoPolo** [comment]
I'm going to merge this since I want to use these types

**Pull request merged**